Sources/SwiftGodot/Core/VariantConvertible.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  Exportable.swift
+//  SwiftGodot
+//
+//  Created by Elijah Semyonov on 08/04/2025.
+//
+
+/// Protocol for types that can be converted to and from ``Variant``.
+/// NOTE: this type is planned to supersede ``VariantStorable`` in the future.
+public protocol VariantConvertible {
+    /// Converts the instance to a ``Variant``.
+    func toVariant() -> Variant
+    
+    /// Extract ``Self`` from a ``Variant``. Returns `nil` if it's not possible. E.g. another type is stored in the `variant`
+    static func fromVariant(_ variant: Variant) -> Self?
+}
+
+/// Internal API. Interface for types that contains details on how it interacts with C GDExtension API.
+/// This is more specialized version of ``VariantConvertible`` for cases where an ability to be converted to and from a ``Variant`` is not enough.
+/// At the same time it allows ``VariantConvertible`` to be implemented by the user for coding arbitary values inside the ``Variant``
+public protocol _GodotBridgeable: VariantConvertible {
```

Comment:
Would love to include in the documentation here the reason for the cases where it needs additional support, as I do not understand it myself now, and would love to have this information in the future.